custom_components/tahoma/climate_devices/somfy_heating_temperature_interface.py

```diff
@@ -0,0 +1,221 @@
+}
+
+
+class SomfyHeatingTemperatureInterface(OverkizEntity, ClimateEntity):
+    """Representation of Somfy Heating Temperature Interface."""
+
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_hvac_modes = [*HVAC_MODES_TO_TAHOMA]
+    _attr_preset_modes = [*PRESET_MODES_TO_TAHOMA]
+    _attr_supported_features = SUPPORT_PRESET_MODE
+
+    def __init__(self, device_url: str, coordinator: OverkizDataUpdateCoordinator):
+        """Init method."""
+        super().__init__(device_url, coordinator)
+
+        self._temp_sensor_entity_id = None
+        self._current_temperature = None
+
+    async def async_added_to_hass(self):
+        """Register temperature sensor after added to hass."""
+        await super().async_added_to_hass()
+
+        # Only the AtlanticElectricarHeater WithAdjustableTemperatureSetpoint has a separate temperature sensor
+        if self.device.widget != "SomfyHeatingTemperatureInterface":
+            return
+
+        entity_registry = await self.hass.helpers.entity_registry.async_get_registry()
+        self._temp_sensor_entity_id = next(
+            (
+                entity_id
+                for entity_id, entry in entity_registry.entities.items()
+                if entry.unique_id == f"{self.base_device_url}#2-core:TemperatureState"
+            ),
+            None,
+        )
+
+        if self._temp_sensor_entity_id:
+            async_track_state_change(
+                self.hass, self._temp_sensor_entity_id, self._async_temp_sensor_changed
+            )
+
+        else:
+            _LOGGER.warning(
+                "Temperature sensor could not be found for entity %s", self.name
+            )
+
+        @callback
+        def _async_startup(event):
+            """Init on startup."""
+            if self._temp_sensor_entity_id:
+                temp_sensor_state = self.hass.states.get(self._temp_sensor_entity_id)
+                if temp_sensor_state and temp_sensor_state.state != STATE_UNKNOWN:
+                    self.update_temp(temp_sensor_state)
+
+        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, _async_startup)
+
+        self.schedule_update_ha_state(True)
+
+    async def _async_temp_sensor_changed(self, entity_id, old_state, new_state) -> None:
+        """Handle temperature changes."""
+        if new_state is None or old_state == new_state:
+            return
+
+        self.update_temp(new_state)
+        self.schedule_update_ha_state()
+
+    @callback
+    def update_temp(self, state):
+        """Update thermostat with latest state from sensor."""
+        if state is None or state.state == STATE_UNKNOWN:
+            return
+
+        try:
+            self._current_temperature = float(state.state)
+        except ValueError as ex:
+            _LOGGER.error("Unable to update from sensor: %s", ex)
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        if OverkizState.CORE_ON_OFF in self.device.states:
```

Comment:
Not sure if this if statement is required, since `select_state` will return None if the state cannot be found?